src/index.ts

```diff
@@ -34,8 +35,10 @@ async function run() {
         (cmd,args)=>exec.exec(cmd,args,execOptions),
         startupOptions
     );
+    buildSummary.processAndDisplayBuildSummary();
 }
 
 run().catch((e) => {
+    buildSummary.processAndDisplayBuildSummary();
```

Comment:
What happens if the failure occurred during `processAndDisplayBuildSummary` in `run`?

Reply:
It show an error message with appropriate stack trace and summary table will not be shown here 

 catch (error) {
    console.error('An error occurred while reading the build summary file or adding the build summary table:', error);
  }

Reply:
I think the challenge here is that if `processAndDisplayBuildSummary` fails on line 38, it's likely to fail on line 42 as well. So you'll get 2 console errors saying the same thing and `core.setFailed` will never be reached.

Reply:
would this be a better option if `processAndDisplayBuildSummary` did not catch any exception ?
```typescript
  buildSummary.processAndDisplayBuildSummary();
    } catch (e) {
        try {
            buildSummary.processAndDisplayBuildSummary();
        } catch (summaryError) {
            core.error(`An error occurred while reading the build summary file or adding the build summary table:', summaryError`);
        }
        core.setFailed(e.message);
    }
}

run().catch((e) => {
    buildSummary.processAndDisplayBuildSummary();
    core.setFailed(e);
});
```

Reply:
@mcafaro does the current changes look good ?